lib/rom/memory/dataset.rb

```diff
@@ -46,9 +46,21 @@ def project(*names)
 
       # Sort a dataset
       #
+      # @param [Array<Symbol>] names
+      #   Names of fields to order tuples by
+      #
+      # @option [Boolean] :nils_first (false)
+      #   Whether `nil` values should be placed before others
+      #
       # @api public
-      def order(*names)
-        sort_by { |tuple| tuple.values_at(*names) }
+      def order(*fields)
+        nils_first = fields.pop[:nils_first] if fields.last.is_a?(Hash)
+
+        sort do |a, b|
+          fields # finds the first difference between selected fields of tuples
+            .map { |n| __compare a[n], b[n], nils_first }
+            .detect { |r| r != 0 } || 0
```

Comment:
One more interesting question. How `order` should behave in case some values cannot be ordered?
```ruby
[
  { name: "Alice",   age: 7 },
  { name: "Knave", age: '1@2'.to_c }
]
```
It could either raise, or remain data unsorted. This needs a decision, I think